opencog/nlp/sureal/surface-realization.scm

```diff
@@ -30,13 +30,22 @@
 )
 
 ; ---------------------------------------------------------------------
+; A wrapper, call SuReal with or without the optional parameter "thoroughness",
+; which is the number of results will be returned by SuRealPMCB
+(define sureal 
+	(case-lambda
+		((setlink) (sureal-main setlink 20))
+		((setlink thoroughness) (sureal-main setlink thoroughness))
+	)
+)
+
 ; A SetLink is the input because it is assumed that the output of the micro-planner
 ; is unordered.
-(define (sureal a-set-link)
+(define (sureal-main a-set-link thoroughness)
```

Comment:
Instead of splitting `sureal` into `sureal` and `sureal-main`, I think they can be in the same function (they are short enough, and will be less confusing).

Also, you have default as 20.  I think the default should be "search completely" (ie. 100%), meaning there should be some way to allow searching 100% (which I don't think is possible with the current implementation yet).